lib/rack/joint/redirect_interface.rb

```diff
@@ -21,8 +20,8 @@ def apply!
         @scheme ||= request_scheme
         @new_host ||= old_host
         @new_path ||= old_path
-        old_url = build_uri(request_scheme, old_host, old_path)
-        new_location = build_uri(scheme, @new_host, @new_path)
+        old_url = UrlBuilder.new(request_scheme, request_query, old_host, old_path).build
```

Comment:
I give request's `query_string` directory to `UrlBuilder` because I have supposed that no user wishes to set query parameters in `config.ru`.